components/TokenInput.tsx

```diff
@@ -1,50 +1,52 @@
-import { useState } from 'react';
+import React from 'react';
+import { UseFormRegisterReturn } from 'react-hook-form';
 import { ChevronDownIcon } from '@chakra-ui/icons';
 import { Input, InputGroup, InputRightElement } from '@chakra-ui/input';
 import { useDisclosure } from '@chakra-ui/react';
 import TokenSelectionModal from '@/components/TokenSelectionModal';
-import { useToken } from '@/contexts/TokenContext';
 import { TokenListContextItem } from '@/contexts/TokenContext';
 
 type TokenInputProps = {
   onSelect: (token: TokenListContextItem) => void; // eslint-disable-line no-unused-vars
-};
+} & UseFormRegisterReturn;
 
-const TokenInput = ({ onSelect }: TokenInputProps) => {
-  const { tokenList } = useToken();
-  const [token, setToken] = useState(tokenList[0]); // Select native token
-  const {
-    isOpen: isTokenSelectionOpen,
-    onOpen: onTokenSelectionOpen,
-    onClose: onTokenSelectionClose,
-  } = useDisclosure();
-  const localOnSelect = (token: TokenListContextItem) => {
-    onSelect(token);
-    onTokenSelectionClose();
-    setToken(token);
-  };
-  return (
-    <>
-      <InputGroup cursor="pointer" onClick={() => onTokenSelectionOpen()}>
-        <Input
-          type="button"
-          pr="4.5rem"
-          size="lg"
-          height="4rem"
-          value={token?.name || tokenList[0]?.name || ''} // Default to native which is assumed at the top of the token list
-          mb=".75rem"
+const TokenInput = React.forwardRef(
+  ({ onSelect, ...rest }: TokenInputProps, ref: React.Ref<HTMLInputElement>) => {
+    const {
+      isOpen: isTokenSelectionOpen,
+      onOpen: onTokenSelectionOpen,
+      onClose: onTokenSelectionClose,
+    } = useDisclosure();
+    const localOnSelect = (token: TokenListContextItem) => {
+      onSelect(token);
+      onTokenSelectionClose();
+    };
+    return (
+      <>
+        <InputGroup cursor="pointer" onClick={() => onTokenSelectionOpen()}>
+          <Input
```

Comment:
think the display might be off a little bit: perhaps we can display the token icon and/or choose a different alignment for the token name text. should I create a new issue?

Reply:
Yea, create a new ticket for that probably. I realigned the carat after looking at this comment.

Reply:
#33 Should be captured here